### PR TITLE
tgi codestral pr

### PR DESCRIPTION
**Pull request by nithiyn**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: notebooks/Codestral-samples/codestral_chat_ui/codestral_chat.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import gradio as gr
+import boto3
```
Comment (tugglep) [inline notebooks/Codestral-samples/codestral_chat_ui/codestral_chat.py:2]: make sure imports are in alphabetical order

**Review by tugglep — CHANGES_REQUESTED** [review]

File: notebooks/Codestral-samples/Deployment-and-Inference-Guide-Codestral-TGI.ipynb [hunk 2]
```diff
@@ -0,0 +1,653 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "<div class=\"alert alert-block alert-warning\"> \n",
+    "\n",
+    "<b>NOTE:</b> Codestral is a 22B open-weight model licensed under the new Mistral AI Non-Production License, which means that you can use it for research and testing purposes. Codestral can be downloaded on HuggingFace.\n",
+    "If you want to use the model in the course of commercial activity, Commercial licenses are also available on demand by reaching out to the team.\n",
```
Comment (tugglep) [inline notebooks/Codestral-samples/Deployment-and-Inference-Guide-Codestral-TGI.ipynb:45]: Please clarify the term 'the team' - I know you mean Mistral, but I would just be super clear here.

File: notebooks/Codestral-samples/Deployment-and-Inference-Guide-Codestral-TGI.ipynb [hunk 3]
```diff
@@ -0,0 +1,653 @@
...
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "\n",
+    "# Sagemaker endpoint configuration\n",
+    "instance_type = \"ml.g5.12xlarge\"   #/g5.48xlarge \n",
+    "number_of_gpus = 4                 #number of gpus the instance in use has\n",
+    "health_check_timeout = 900         #additional time to load in the model\n",
```
Comment (tugglep) [inline notebooks/Codestral-samples/Deployment-and-Inference-Guide-Codestral-TGI.ipynb:301]: is it worth talking about health_check_timeout ? what happens if the timeout is 500? or 1000+

File: notebooks/Codestral-samples/Deployment-and-Inference-Guide-Codestral-TGI.ipynb [hunk 4]
```diff
@@ -0,0 +1,653 @@
...
+    "max_total_tokens = 32768           #max total tokens for codestral\n",
+    "\n",
+    "\n",
+    "config = {\n",
+    "  'HF_MODEL_ID': \"mistralai/Codestral-22B-v0.1\",                                    # model_id from HuggingFace\n",
+    "  'HF_TASK': \"text-generation\",                                                     # huggingface inference pipeline\n",
+    "  'SM_NUM_GPUS': json.dumps(number_of_gpus),                                        # Number of GPU used per replica\n",
+    "  'HUGGING_FACE_HUB_TOKEN': \"<REPLACE WITH YOUR TOKEN>\",                            #add your huggingface hub access token with read permissions\n",
```
Comment (tugglep) [inline notebooks/Codestral-samples/Deployment-and-Inference-Guide-Codestral-TGI.ipynb:309]: Did we say we need a HF token before this point? (I don't remember) if we don't, we should say upfront the user needs one.

Comment (nithiyn) [reply notebooks/Codestral-samples/Deployment-and-Inference-Guide-Codestral-TGI.ipynb:309]: resolved

File: notebooks/Codestral-samples/Deployment-and-Inference-Guide-Codestral-TGI.ipynb (hunk 3 above)
Comment (nithiyn) [reply notebooks/Codestral-samples/Deployment-and-Inference-Guide-Codestral-TGI.ipynb:301]: done

File: notebooks/Codestral-samples/Deployment-and-Inference-Guide-Codestral-TGI.ipynb (hunk 2 above)
Comment (nithiyn) [reply notebooks/Codestral-samples/Deployment-and-Inference-Guide-Codestral-TGI.ipynb:45]: done

File: notebooks/Codestral-samples/codestral_chat_ui/codestral_chat.py (hunk 1 above)
Comment (nithiyn) [reply notebooks/Codestral-samples/codestral_chat_ui/codestral_chat.py:2]: done

**Pull request merged**